### DISCUSSION TITLE
How to use ignore-methods

### DISCUSSION THREAD

**Discussion by kreghek** [stryker-mutator/stryker-net#1478] [Q&A] [answered]
I use Stryker to test my project. And I want to exclude some "auto-generated" methods like ToString and GetHashCode.

The command to run Stryker looks like:
`-ignore-methods "['ToString', 'ConfigureAwait', '*Exception.ctor', 'Equals', 'GetHashCode']"`

But I has no effect:
![image](https://user-images.githubusercontent.com/2405499/111724980-c5cec100-8898-11eb-847b-ab3d6ffe1f5b.png)

**Expected behavior**
The methods are ignored and don't appears in report.

**Desktop (please complete the following information):**
 - OS: ubuntu-latest
 - Type of project: .NET Standart
 - Framework Version 2.0
 - Stryker Version 0.22.0

**Comment by rouke-broersma** [comment, marked as answer]
Hi @kreghek

The ignore methods option only ignores methods **calls**, not method bodies. For method bodies you'll want to use the ignore code coverage attribute.

**Reply by aurnoi1** [reply]
Hi @rouke-broersma ,

I am not sure which "[ignore coverage attribute](https://stryker-mutator.io/docs/stryker-net/Configuration/)" you make reference too. May you provide an example please? I have a method (GetFeatures) covered by functional tests and I would like to exclude it from stryker.

ex:
```c#
        public List<Feature> GetFeatures(DirectoryInfo directoryInfo, string conditions)
        {
            var gherkinParser = new GherkinParser();
            var matches = new List<Feature>();
            foreach (var file in directoryInfo.GetFiles("*.feature", SearchOption.AllDirectories))
            {
                var gherkinDocument = new gherkin.Parser().Parse(file.FullName);
                if (gherkinParser.GetScenarios(gherkinDocument, conditions).Count() > 0)
                {
                    matches.Add(gherkinDocument.Feature);
                }
            }

            return matches;
        }
```

Thanks for your support!

**Reply by aurnoi1** [reply]
Ouch... I just understood that the attribute is the "System.Diagnostics.CodeAnalysis.ExcludeFromCodeCoverageAttribute".
It does the job!

```csharp
using System.Diagnostics.CodeAnalysis;

[ExcludeFromCodeCoverage]
public List<Feature> GetFeatures(DirectoryInfo directoryInfo, string conditions)
{
   ...
}
```

Thanks
